### PR TITLE
Add more quiz

### PR DESCRIPTION
**Pull request by Zeegaths**
#729 Added three quizzes to chapter 3 - Arrays

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/732)

### CODE REVIEW

**Comment by TAdev0** [comment]
Hi @Zeegaths ! 
Your PR is empty, can you check again your last commit and push it correctly?

**Comment by Zeegaths** [comment]
@TAdev0  done. sorry for the oversight

**Review by TAdev0 — CHANGES_REQUESTED** [review]

File: quizzes/ch03-01-arrays.toml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+
+
+
+[[questions]]
+id = "0xcccccc"
+type = "MultipleChoice"
+prompt.prompt = """
+What happens when you run this function?
```
[the remark this answers is not in the document]
Comment (Zeegaths) [reply quizzes/ch03-01-arrays.toml:46]: I've made the changes

**Comment by enitrat** [comment]
@Zeegaths The goal of quizzes is to challenge the reader on its understanding of the material. We do not want to ask simple questions about the reading capacities of the reader, we want to challenge its comprehension. As such, we need to orient our questions in this purpose

What is useful:
- Asking questions about "deep" mechanisms: e.g., 
  - what would make a an array access panic - wrapped in a layer of obsfucation and with distractors, e.g. a function that would search for a value in up to `n` elements in an array, 
  - verifying that the immutability mechanism is understood properly 
  - verifying that the user understood what the array represents in memory: what happens if it's "owned" by two variables?

What is not useful:
- Asking questions about surface stuff (how to access an array's element - this is _trivial_). or, maximum __one__
- Asking questions not related to arrays directly (e.g. your two last question are not related to arrays)

**Comment by Zeegaths** [comment]
Ok got it

File: quizzes/ch03-01-arrays.toml [hunk 2]
```diff
@@ -2,43 +2,33 @@
...
+    a.append(3);
+
+    let first_value = a.pop_front().unwrap();
+
+    println!("The first value is {}", first_value);  
 }
 ```
 """
-answer.answer = "`arr.pop_front()`"
+answer.answer = "It panics"
```
Comment (enitrat) [inline quizzes/ch03-01-arrays.toml]: This is not true, your program doesn't even compile.

**Comment by Zeegaths** [comment]
I've added quizzes targeting mutability and access. lmk if i should make more changes

**Comment by enitrat** [comment]
@Zeegaths I advise you spend some more time studying and understanding how Cairo works, this will make it easier to propose quizzes that are challenging. With a little more experience you will be able to also identify what you have been struggling with and take inspiration from some questions.

File: quizzes/ch03-01-arrays.toml (hunk 2 above)
Comment (Zeegaths) [reply quizzes/ch03-01-arrays.toml]: so it panics? or should i explicitly state it doesn't compile

**Comment by Zeegaths** [comment]
ok

File: quizzes/ch03-01-arrays.toml (hunk 2 above)
Comment (enitrat) [reply quizzes/ch03-01-arrays.toml]: No, the example is not correct for what you are trying to demonstrate

**Comment by Zeegaths** [comment]
I think i had the wrong commit.., I have updated it, please check it out

**Comment by Zeegaths** [comment]
Thank you for the clarification. I have made the fixes

**Comment by enitrat** [comment]
https://github.com/cairo-book/cairo-book/pull/732#issuecomment-2061219239 Please take my comment into consideration, I even added specific examples of what you could do. What you're proposing for now is not of good quality

**Comment by Zeegaths** [comment]
Thanks for the feedback.

**Pull request closed without merging**